internal/verifier/clustertime.go

```diff
@@ -0,0 +1,141 @@
+package verifier
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/10gen/migration-verifier/internal/logger"
+	"github.com/10gen/migration-verifier/internal/retry"
+	"github.com/10gen/migration-verifier/internal/util"
+	"github.com/10gen/migration-verifier/mbson"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+const opTimeKeyInServerResponse = "operationTime"
+
+// GetClusterTime returns the remote cluster’s cluster time.
+// In so doing it creates a new oplog entry across all shards.
+// The “note” will go into the cluster’s oplog, so keep it
+// short but meaningful.
+func GetClusterTime(
+	ctx context.Context,
+	logger *logger.Logger,
+	client *mongo.Client, // could change to type=any if we need
```

Comment:
I don't understand this comment; surely we must always have a mongo client, because we're calling mongo-shaped things on it!